frontend/rust-lib/flowy-database2/src/services/database_view/view_editor.rs

```diff
@@ -508,6 +530,78 @@ impl DatabaseViewEditor {
     Ok(())
   }
 
+  pub async fn v_get_all_calculations(&self) -> Vec<Arc<Calculation>> {
+    self.delegate.get_all_calculations(&self.view_id)
+  }
+
+  pub async fn v_update_calculations(
+    &self,
+    params: UpdateCalculationChangesetPB,
+  ) -> FlowyResult<()> {
+    let calculation_id = match params.calculation_id {
+      None => gen_database_calculation_id(),
+      Some(calculation_id) => calculation_id,
+    };
+
+    let calculation = Calculation {
+      id: calculation_id,
+      field_id: params.field_id,
+      calculation_type: params.calculation_type.value(),
+      value: "".to_owned(), // Empty for non-calculated
+    };
+
+    let calculations_controller = self.calculations_controller.clone();
+    let changeset = calculations_controller
+      .did_receive_changes(CalculationChangeset::from_insert(calculation.clone()))
+      .await;
+
+    if let Some(changeset) = changeset {
+      if !changeset.insert_calculations.is_empty() {
+        for insert in changeset.insert_calculations.clone() {
+          let calculation: Calculation = Calculation::from(&insert);
+          self
+            .delegate
+            .update_calculation(&params.view_id, calculation);
+        }
+      }
+
+      notify_did_update_calculation(changeset).await;
+    }
+
+    drop(calculations_controller);
+
+    Ok(())
+  }
+
+  pub async fn v_remove_calculation(
+    &self,
+    params: RemoveCalculationChangesetPB,
+  ) -> FlowyResult<()> {
+    self
+      .delegate
+      .remove_calculation(&params.view_id, &params.calculation_id);
+
+    let calculation = Calculation {
+      id: params.calculation_id.clone(),
+      field_id: params.field_id.clone(),
+      calculation_type: 0,
+      value: "".to_owned(),
+    };
+
+    let calculations_controller = self.calculations_controller.clone();
```

Comment:
No need to clone the `calculations_controller`. Just use `self. calculations_controller.did_receive_changes`

Reply:
You can cd rust-lib and then run 
```shell
cargo clippy --fix --allow-dirty 
```

or just check
```shell
cargo clippy --all-targets -- -D warnings
```